### PR TITLE
focus first textbox element

### LINKED ISSUE #6247: Element focus
Rocket.Chat Version: 0.53.0
Running Instances: 1
DB Replicaset OpLog: Enabled
Node Version: 4.7.3 - x64

While creating new channels by clicking `+` icon, input element of a rendered form is not focused. 

focus

### LINKED ISSUE #6248: Leverage ESC key for better UX
Rocket.Chat Version: 0.53.0
Running Instances: 1
DB Replicaset OpLog: Enabled
Node Version: 4.7.3 - x64

ESC key is used to clear the search bar which is good for user experience. 

There are certain areas where we can use ESC key for better UX. 
1. Flex Tab - Use ESC key to hide contents instead of clicking flex tab button for the second time.
2. Side Nav - Use ESC key to get previous UI state.
3. Video Dialog - Use ESC key to cancel video dialog.
4. Alert Dialog - Use ESC key to cancel alerts.

### PR DESCRIPTION
**Pull request by a5his**
@RocketChat/core 

While creating new channels by clicking + icon, first textbox input element of a rendered form gets focus.

Closes #6247
Closes #6248 

focustextbox

escape

### CODE REVIEW

File: packages/rocketchat-ui/client/lib/esc.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+		if (flextTabButton) {
+			flextTabButton.dispatchEvent(this.getClickEvent());
+			return;
+		}
+	},
+
+	sideNavIcon() {
+		const sideNavArrow = document.querySelector('.side-nav .arrow');
+		if (sideNavArrow && (sideNavArrow.classList.contains('top') || sideNavArrow.classList.contains('close'))) {
+			sideNavArrow.dispatchEvent(this.getClickEvent());
```
Comment (karlprieb) [inline packages/rocketchat-ui/client/lib/esc.js]: I think you can use `SideNav.toggleCurrent()` here. What you think?

File: packages/rocketchat-ui/client/lib/esc.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+			return;
+		}
+	},
+
+	videoDialog() {
+		const vrecDialog = document.querySelector('.vrec-dialog');
+		if (vrecDialog && Number(window.getComputedStyle(vrecDialog).opacity) === 1) {
+			document.querySelector('.vrec-dialog .button.cancel').dispatchEvent(this.getClickEvent());
```
Comment (karlprieb) [inline packages/rocketchat-ui/client/lib/esc.js]: you can use 
```
VideoRecorder.stop();
VRecDialog.close();
```
here

File: packages/rocketchat-ui/client/lib/sideNav.coffee [hunk 3]
```diff
@@ -36,8 +36,18 @@
 			sideNav.find("header").removeClass "hover"
 
 	focusInput = ->
+		sideNavDivs = _.filter document.querySelectorAll('aside.side-nav')[0].children, (ele) ->
+			ele.tagName == 'DIV' and !ele.classList.contains('hidden')
+		highestZidx = 0
+		highestZidxElem = undefined
+		_.each sideNavDivs, (ele) ->
+			zIndex = window.getComputedStyle(ele).zIndex
+			if Number(zIndex).toString() != 'NaN' and Number(zIndex) > highestZidx
```
Comment (karlprieb) [inline packages/rocketchat-ui/client/lib/sideNav.coffee]: I think you can set `zIndex = Number(window.getComputedStyle(ele).zIndex)` and in this if just check `if zIndex > highestZidx`. NaN is false :) so NaN > any number will be false

File: packages/rocketchat-ui/client/lib/esc.js (hunk 1 above)
Comment (a5his) [inline packages/rocketchat-ui/client/lib/esc.js]: 👍

File: packages/rocketchat-ui/client/lib/esc.js (hunk 2 above)
Comment (a5his) [inline packages/rocketchat-ui/client/lib/esc.js]: 👍

File: packages/rocketchat-ui/client/lib/sideNav.coffee (hunk 3 above)
Comment (a5his) [inline packages/rocketchat-ui/client/lib/sideNav.coffee]: You're right. 👍

**Pull request merged**